### PR TITLE
- Add `chk_class()` to check inheritance ignoring class type.

### PR DESCRIPTION
**Pull request by StefanoMezzini**
Based on issue https://github.com/poissonconsulting/chk/issues/243, I suggest we use a generic `check_class()` that is exactly like `check_s3_class()` but does not check for the object not being S4. See the diff at https://www.diffchecker.com/zDl0S2ID/. Additionally, `!isS4()` no longer guarantees that the class is S3, since it could also be S6.

Changes include:

- created `chk_class()` as detailed above,
- substituted `chk_s3_class()` with `chk_class()` in `check_values()`
- updated the docs with the local RStudio build & check routine
- added tests for `chk_class()`
- errors for `chk_s3_class()` and `chk_s4_class()` now specify what the object's class(es) is (are) to help diagnose issues
- vectorized `vld_s4_class()` to check for any of the classes passed, similarly to `vld_s3_class()`. It previously only accepted one class.
- updated tests accordingly.
- (added the AI gitignore stuff that is now in all repos)

### CODE REVIEW

**Review by joethorley — CHANGES_REQUESTED** [review]
It looks good. I like the updates to check_values() and check_data(). We just need some directly testing of `vld_inherits()` and `chk_inherits()` including edge cases. You can look at the existing tests for inspiration.

**Comment by StefanoMezzini** [comment]
do you want me to create `vld_inherits()` and `chk_inherits()`?

**Review by joethorley — DISMISSED** [review]
Looks good

File: .Rbuildignore [hunk 1]
```diff
@@ -22,3 +22,5 @@
 ^\.vscode$
 ^[.]?air[.]toml$
 ^\.ccache$
```
Comment (joethorley) [inline .Rbuildignore:24]: merged

**Review by joethorley — DISMISSED** [review]

**Review by aylapear — COMMENTED** [review]
I don't have any changes on the code, it all looks good. Just a question to get a response before I provide approval is are there any specific tests for `chk_class()`. I see it is being tested through `check_values()` but there is not test file for itself.

**Comment by StefanoMezzini** [comment]
Added tests targeting `chk_class()` directly with commit 6e25b1e923b6b95a7bf0dcdd6f32aeed35666232. The function has full test coverage from the new tests:
```
covr::report(covr::file_coverage("R/chk-class.R", "tests/testthat/test-chk-class.R"))
```
image

**Comment by StefanoMezzini** [comment]
Closes #243, as the issues have been resolved:

```

> library(chk)
> check_values("01:02:03", "")
> class(lubridate::hms("01:02:03"))
[1] "Period"
attr(,"package")
[1] "lubridate"
> ## [1] "Period"
> ## attr(,"package")
> ## [1] "lubridate"
> check_values(lubridate::hms("01:02:03"), lubridate::hms("01:02:03"))
```

image

**Review by aylapear — DISMISSED** [review]
Thanks for adding the tests and checking the coverage!

**Comment by StefanoMezzini** [comment]
I had forgotten to add some changes to the errors that we had discussed previously. The error messages for `chk_s3_class()` and `chk_s4_class()` now specify what the object's class is to help diagnose issues (6106f23). The syntax is similar to that of other functions:
```
> chk_gt(-1)
# ! `-1` must be greater than 0, not -1.
> chk_gte(-1)
# ! `-1` must be greater than or equal to 0, not -1.> cc(1, brac = character(0))
> cc(1, brac = character(0))
# ! `length(brac)` must be between 1 and 2, not 0.
> chk_lt(1)
# ! `1` must be less than 0, not 1.
> chk_lte(1)
# ! `1` must be less than or equal to 0, not 1.
> chk_range(2)
# ! `2` must be between 0 and 1, not 2.
> chk_subset(1, 0)
# ! `1` must match 0, not 1.
> chk_subset(1, c(0, 2))
# ! `1` must match 0 or 2, not 1.
```

```
> chk_s3_class(1, "integer")
Error:
! `1` must inherit from S3 class 'integer', not S3 class 'numeric'.
> chk_s3_class(matrix(1), "numeric")
Error:
! `matrix(1)` must inherit from S3 class 'numeric', not S3 class 'matrix', 'array'.
```

~Note the singular "class" even when there are multiple classes.~ fixed: see comment below

I've also allowed `vld_s4_class()` to check for any of the classes passed (7677a83), similarly to `vld_s3_class()`. It previously only accepted one class.

Finally, I've updated the tests accordingly.

**Comment by StefanoMezzini** [comment]
I've improved the handling of plurals when there are multiple classes in c67b795 and 5528ac5. But see https://github.com/poissonconsulting/chk/issues/293 for updating the `%` syntax.
```
> chk_s3_class(foo, c("b", "c"))
Error:
! `foo` must inherit from S3 classes 'b' or 'c', not S3 classes 'a' and 'z'.
```

**Review by joethorley — APPROVED** [review]

**Review by aylapear — APPROVED** [review]
Good job on catching the plural error messaging and building tests for it as well.
